platform/innovium/sonic-platform-modules-cameo/esc601-32q/modules/at24_smbus.c

```diff
@@ -0,0 +1,847 @@
+/*
+ * at24.c - handle most I2C EEPROMs
```

Comment:
linux kernel has at24 driver. why not use the original one?

Reply:
Hi Iguohan,
     we are using at24c16 on our board, 24c16 is 16bit address mode.  In current driver at24.c.
     it will check I2c Function bit with 16bit address mode.  our cpu's I2c bus is SMBUS, but 24c16 need I2C.  so we changed it.
	_if (!i2c_check_functionality(client->adapter, I2C_FUNC_I2C)) {
		//if (chip.flags & AT24_FLAG_ADDR16)
		//	return -EPFNOSUPPORT;_

Reply:
the same issue with  mcp3422

Reply:
Our CPU is using SMBUS not pure I2C, therefore we have to marked out the functionality check. Otherwise it shows us init checking error. No other company with the same issue? I mean that some minor change required with native driver.